doc/usecases/core/UC_keyset_lookup_prefix.md

```diff
@@ -0,0 +1,25 @@
+# Use Case: Lookup hierarchy in `KeySet`
+
+## Summary
+
+- **Title:** Lookup hierarchy in `KeySet`
+- **Scope:** `libelektra-core`
+- **Level:** Developer Goal
+- **Actors:** Core, Caller
+- **Brief:** Caller looks up `Key` hierarchy in existing `KeySet`
```

Comment:
Not clear to me what this means. I am not sure if we need this usecase. Please also add an example.

Reply:
> Not clear to me what this means. I am not sure if we need this usecase.

It's basically `ksFindHierarchy`, which is a slightly restricted kind of "prefix lookup", i.e., finding all items that start with a given prefix. The restriction is that the prefix must end with a zero-terminator that is significant for the lookup, i.e., we only find keys that are below. Although I did not and do not want to use that terminology, because that requires extra Elektra-specific explanations.

> Please also add an example.

Where, there is not really a place for this in the template. Maybe more explanation will help instead?